### DISCUSSION TITLE
Loading module/signing the module

### DISCUSSION THREAD

**Discussion by sunilchintu2468** [microsoft/azurelinux#4797] [Q&A]
Bought up Mariner VM  (Mariner-1 latest ISO), on a CentOS host operating system.
And tried to compile and load a simple kernel module, but unfortunately due to the kernel configuration (CONFIG_MODULE_SIG=y), facing an error in loading the compiled module.

Now to sign this module, followed the steps mentioned at [](https://www.kernel.org/doc/html/v5.9/admin-guide/module-signing.html).

Unfortunately, facing an error in adding the key:
root@cbl-mariner-UjcaFoEMLXRM [ /home/sunil/testModule ]# cat /proc/keys | grep builtin
10e8a250 I------     1 perm 1f0b0000     0     0 keyring   .builtin_trusted_keys: 2
root@cbl-mariner-UjcaFoEMLXRM [ /home/sunil/testModule ]# sudo keyctl padd asymmetric "" 0x10e8a250  < MOK.der
add_key: Permission denied

I tried to check this error on the internet, but no luck. Can you please help?

**Comment by PawelWMS** [comment]
Hey, @sunilchintu2468,

I am not familiar with the `keyctl` approach to adding a new trusted signing key, but if you only want to play around with custom kernel modules and don't care about their signatures being verified, you can disable that verification inside `/boot/grub2/grub.cfg` by removing `lockdown=integrity` from kernel's input arguments. You'd need to reboot the system after that.
